h5pxblock/static/js/src/h5pxblock.js

```diff
@@ -1,85 +1,131 @@
+            {
+              state: args.userData,
+            },
+          ],
+          ajax: {
+            contentUserDataUrl: contentUserDataUrl,
+          },
+        };
 
-    const playerPromise = function edXH5PPlayer(el) {
-        if (el && $(el).children('.h5p-iframe-wrapper').length == 0) {
-            const userObj = { 'name': args.user_full_name, 'mail': args.user_email };
-            const options = {
-                h5pJsonPath: args.h5pJsonPath,
-                frameJs: 'https://cdn.jsdelivr.net/npm/h5p-standalone@3.6.0/dist/frame.bundle.js',
-                frameCss: 'https://cdn.jsdelivr.net/npm/h5p-standalone@3.6.0/dist/styles/h5p.css',
-                frame: args.frame,
-                copyright: args.copyright,
-                icon: args.icon,
-                fullScreen: args.fullScreen,
-                user: userObj,
-                saveFreq: args.saveFreq,
-                customJs: args.customJsPath,
-                contentUserData: [{
-                    state: args.userData
-                }],
-                ajax: { 
-                    contentUserDataUrl: contentUserDataUrl
-                }
+        try {
+          await new H5PStandalone.H5P(el, options);
+          $(el).siblings(".spinner-container").find(".spinner-border").hide();
+          $(el).show();
+
+          H5P.externalDispatcher.on("xAPI", (event) => {
+            let hasStatement = event && event.data && event.data.statement;
+            if (!hasStatement) {
+              return;
             }
 
-            return new H5PStandalone.H5P(el, options).then(function(){ 
-                $(el).siblings('.spinner-container').find('.spinner-border').hide();
-                $(el).show();        
-                H5P.externalDispatcher.on("xAPI", (event) => {
+            let statement = event.data.statement;
+            let validVerb =
+              statement.verb &&
+              statement.verb.display &&
+              statement.verb.display["en-US"];
+            if (!validVerb) {
+              return;
+            }
+
+            let isCompleted =
+              statement.verb.display["en-US"] === "answered" ||
+              statement.verb.display["en-US"] === "completed";
+            let isChild =
+              statement.context &&
+              statement.context.contextActivities &&
+              statement.context.contextActivities.parent &&
+              statement.context.contextActivities.parent[0] &&
+              statement.context.contextActivities.parent[0].id;
+
+            // Store only completed root events.
+            if (isCompleted && !isChild) {
+              $.ajax({
+                type: "POST",
+                url: contentxResultSaveUrl,
+                data: JSON.stringify(event.data.statement),
+              })
+              .done(function () {
+                // handle fine request  here
+              })
+              .fail(function () {
+                // handle fails request here
```

Comment:
```
              .done(function () {
                // handle fine request  here
              })
              .fail(function () {
                // handle fails request here
```
In the AJAX request, you have comments in the .done() and .fail() handlers, but you don't actually do anything in those handlers. Consider adding appropriate error handling, such as displaying messages to the user or logging errors.

Reply:
This PR is just refactoring the overall code so that it can accept the required h5p module as a parameter so it can be loaded when needed.

This PR doesn't try to refactor the whole code, just provides the minimal required changes to work in the CMS view